playbooks/configuration_project_sync.yml

```diff
@@ -0,0 +1,28 @@
+---
+- hosts: localhost
+  gather_facts: no
+  tasks:
+  - include_vars:
+     ./roles/cluster/defaults/main.yml
+  - include_vars:
+     ./roles/tower/defaults/main.yml
+  - name: Set tower credentials
+    set_fact:
```

Comment:
Rather than setting these tower facts and declaring them in each tower-cli/tower module task, you could just import the `authenticate_tower.yml` playbook at the start of each job template playbook. As the `tower-environment` var is already specified, the `authenticate_tower.yml` playbook should just work as is.

See example:
https://github.com/integr8ly/ansible-tower-configuration/blob/master/playbooks/bootstrap_integreatly_install.yml#L2

Reply:
Ah you're right, I never thought of importing the playbook, that gets around the issue of persisting the login and is cleaner, i'll make that change.